### PR TITLE
UX-951/better errors

### PR DESCRIPTION
**Pull request by jruddell**
#### Cluster Errors Tab
image

#### Platform9 Events Tab
image

#### Notification Detail Modal
image

image

### CODE REVIEW

File: src/app/core/components/notifications/notification-details.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        </SimpleLink>
+      </header>
+      <div className={classes.hr} />
+      {nextViewType !== NotificationViewTypes.Parsed && (
+        <article>
+          <DisplayKeyValues keyValuePairs={upperValues} />
+          <div className={classes.hr} />
+          <DisplayKeyValues keyValuePairs={lowerValues} />
+          <CopyToClipboard
+            header="Api Resonse"
```
Comment (knnguy) [inline src/app/core/components/notifications/notification-details.tsx:105]: Typo

**Review by knnguy — APPROVED** [review]

File: src/api-client/ApiClient.ts [hunk 2]
```diff
@@ -139,41 +179,68 @@ class ApiClient {
...
+          config: formatResponseConfigPayload(responseConfig),
+          data: responseData,
+          status: responseStatus,
+          statusText: responseStatusText,
+        }
 
         // If data contains user password, remove it
-        if (cfg?.data?.includes('{"methods":["password"]')) {
-          const dataObj = JSON.parse(cfg.data)
+        if (responseConfig?.data?.includes('{"methods":["password"]')) {
```
Comment (kevinylee1991) [inline src/api-client/ApiClient.ts:213]: Just a note, there may be other sensitive info fields ex. with VMs with cloudinit, passwords may be sent through there, possibly other API calls as well that I can't think of off the top of my head. Perhaps sensitive keys may need to be stored in the api method metadata?

Also I can't tell, is the responseConfig data the request body that was sent or is that something else?

File: src/app/core/components/notifications/NotificationsPage.tsx [hunk 3]
```diff
@@ -1,58 +1,224 @@
...
+            notificationType={ErrorNotificationType.Platform9Event}
+          />
+        </Tab>
+      </Tabs>
+    </>
   )
 }
 
 export default NotificationsPage
+
```
Comment (kevinylee1991) [inline src/app/core/components/notifications/NotificationsPage.tsx:205]: still need this?

File: src/api-client/Appbert.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ class Appbert extends ApiService {
     return this.client.keystone.getServiceEndpoint('appbert', 'admin')
   }
 
-  @trackApiMethodMetadata({ url: '/clusters', type: 'GET' })
+  @trackApiMethodMetadata({ url: '/clusters', type: 'GET', error: { isClusterError: false } })
```
Comment (kevinylee1991) [inline src/api-client/Appbert.ts:17]: Preferable to have isClusterError: false declared or leave empty? I noticed some in qbert (ex. qbert's /clusters) were left alone

**Review by kevinylee1991 — APPROVED** [review]
looks good

File: src/api-client/Appbert.ts (hunk 4 above)
Comment (jruddell) [reply src/api-client/Appbert.ts:17]: It doesn't make much of a difference, false and undefined are treated the same in a falsey check

File: src/api-client/ApiClient.ts (hunk 2 above)
Comment (jruddell) [reply src/api-client/ApiClient.ts:213]: If you can draw up an example we can think about extending the model to include something like sensitive keys. We can also just have this be global and not specific to a single api method's metadata

File: src/app/core/components/notifications/NotificationsPage.tsx (hunk 3 above)
Comment (jruddell) [reply src/app/core/components/notifications/NotificationsPage.tsx]: nope :)

**Pull request merged**